Documentation/gettingstarted/tls-visibility.rst

```diff
@@ -0,0 +1,308 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+.. _gs_tls_inspection:
+
+***************************************************
+Inspecting TLS Encrypted Communications with Cilium
+***************************************************
+
+This document serves as an introduction for how network security teams can use Cilium to transparently inspect
+TLS-encrypted connections.  This TLS-aware inspection allows Cilium API-aware visibility and policy to function
+even for connections where client to server
+communication is protected by TLS, such as when a client accesses the API service via HTTPS.  This capability is similar to
+what is possible to traditional hardware firewalls, but is implemented entirely in software on the Kubernetes worker node,
+and is policy driven, allowing inspection to target only selected network connectivity.
+
+This type of visibility is
+extremely valuable to be able to monitor how external API services are being used,
+for example, understanding which S3 buckets are being accessed by an given application.
+
+.. include:: gsg_requirements.rst
+
+Edit the ClusterRole for Cilium to give it access to Kubernetes secrets
+
+
+.. parsed-literal::
+
+    $ kubectl edit clusterrole cilium -n kube-system
+
+Add the following section at the end of the file:
+
+
+.. parsed-literal::
+
+  - apiGroups:
+    - ""
+    resources:
```

Comment:
we can add resourceNames to limit scope of secrets cilium should be able to `get`. It still will be `any namespace`, but only when name matches.
```
- apiGroups:
  - ""
  resourceNames:
  - t2ls-orig-data
  resources:
  - secrets
  verbs:
  - get
```

Reply:
I think we discussed this offline, but unless we know exactly the name of all k8s-secrets that will be created for each destination to be terminated, I don't think we can assume we know the names.  Does this "resourceNames" model support regex?  If so, we could perhaps suggest a convention that prefixes secrets with "cilium-"